in_toto/in_toto_run.py

```
@@ -102,37 +99,53 @@ def main():
 
   in_toto_args.add_argument("-m", "--materials", type=str, required=False,
       nargs='+', help="Files to record before link command execution")
+
   in_toto_args.add_argument("-p", "--products", type=str, required=False,
       nargs='+', help="Files to record after link command execution")
 
   in_toto_args.add_argument("-k", "--key", type=str, required=True,
       help="Path to private key to sign link metadata (PEM)")
 
-  in_toto_args.add_argument("-b", "--record-byproducts", dest='record_byproducts',
+  in_toto_args.add_argument("-b", "--record-byproducts",
       help="If set redirects stdout/stderr and stores to link metadata",
-      default=False, action='store_true')
+      dest="record_byproducts", default=False, action="store_true")
+
+  in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+      help="Verbose execution.", default=False, action="store_true")
 
-  link_args = parser.add_argument_group("Link command")
   # FIXME: This is not yet ideal.
   # What should we do with tokens like > or ; ?
-  link_args.add_argument("link_cmd", nargs="+",
+  in_toto_args.add_argument("link_cmd", nargs="+",
     help="Link command to be executed with options and arguments")
 
   args = parser.parse_args()
 
-  in_toto_run(args.step_name, args.key, args.materials, args.products,
-      args.link_cmd, args.record_byproducts)
+  # Turn on all the `log.doing()` in the library
+  if args.verbose:
+    log.logging.getLogger().setLevel(log.logging.INFO)
+
+  # We load the key here because it might prompt the user for a password in
+  # case the key is encrypted. Something that should not happen in the library.
+  try:
+    key = util.prompt_import_rsa_key_from_file(args.key)
+  except Exception as e:
+    log.error("in load key - {}".format(args.key))
+    sys.exit(1)
+
+  in_toto_run(args.step_name, args.materials, args.products,
+      args.link_cmd, key, args.record_byproducts)
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```

This is the CLI tool right? when we log.error, does it pint to the terminal, or is it silently failing right now?

---

Is this question related to the quotes in L150? 

Whether `log.error` prints or not depends on the logging level specified in `settings.py` and if the `--verbose` cli option is specified, which overrides the level in `settings.py` to print everything greater than `logging.INFO`.

---

Maybe you want to add you thoughts on this to https://github.com/in-toto/in-toto/issues/6?

---

No fix needed

---

my question is: what happens if the logger's handler is a file handler?

---

Well, right now the logger does not have file handler. And it is on us to decide how the logger is used. :)

I'd like to improve logging/user feedback in general (cf. https://github.com/in-toto/in-toto/issues/6). E.g. properly separate logging (which could use any handler) and user feedback (which must print to the terminal). We have discussed this before and decided that python's `logging` module can be used for both (given proper namespacing).